### DISCUSSION TITLE
Script to create large data nodes

### DISCUSSION THREAD

**Discussion by ShahanaFarooqui** [lightningnetwork/lnd#6867] [Ideas]
Is it possible to provide a script which can generate large transactions data for a node?

I do not have much data on my node to test my application's performance for a larger node. The ability to generate a node with a lot of peers, channels, transactions can be very helpful in debugging performance issues & errors.

**Comment by guggero** [comment]
I think channels and transactions can probably be created quite easily with a script on regtest.
Peers might be a bit trickier, because that would mean running multiple instances of `lnd` at the same time, which will be limited by the number of free ports on a system.
Our integration test framework might be something to look at to build this. But I'm not aware of anyone who's already done this yet.

For a large number of payments, there's https://github.com/bottlepay/lightning-benchmark.
